process/block/metablock.go

```diff
@@ -1643,12 +1643,35 @@ func (mp *metaProcessor) getLastFinalizedMetaHashForShard(shardHdr *block.Header
 			continue
 		}
 
-		lastFinalizedMetaHash = oldShardHdr.MetaBlockHashes[len(oldShardHdr.MetaBlockHashes)-1]
+		numAddedMetas = len(oldShardHdr.MetaBlockHashes)
```

Comment:
I think that lines between 1646-1674 can be replaced with this shorter code:

		numAddedMetas = len(oldShardHdr.MetaBlockHashes)

		if len(lastMetaHash) == 0 {
			lastMetaHash = oldShardHdr.MetaBlockHashes[numAddedMetas-1]
		} else {
			if !bytes.Equal(lastMetaHash, oldShardHdr.MetaBlockHashes[numAddedMetas-1]) {
				lastFinalizedMetaHash = shardHdr.MetaBlockHashes[numAddedMetas-1]
				return lastMetaHash, lastFinalizedMetaHash, nil
			}
		}

		if numAddedMetas > 1 {
			lastFinalizedMetaHash = shardHdr.MetaBlockHashes[numAddedMetas-2]
			return lastMetaHash, lastFinalizedMetaHash, nil
		}
		
		currentHdr = oldShardHdr

Reply:
there is still an edge case if using the proposed code. When one meta is finished in one shard header, and the next one is the previous one. For that case lastFinalizedMetaHash has to be exactly oldShardHdr.MetaBlockHashes[numAddedMetas-1].